policy/compiler/compiler.go

```diff
@@ -42,38 +43,165 @@ type Compiler struct {
 	reg Registry
 }
 
+// CompileInstance type-checks and validates a parsed representation of a policy instance whose
```

Comment:
Does this reference the validator in the template? Given the variety of validation categories, it would be better to be more concise in the comment.